src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ActivatorUtilities.cs

```diff
@@ -32,10 +31,11 @@ public static class ActivatorUtilities
             [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] Type instanceType,
             params object[] parameters)
         {
-            Type[] argumentTypes = parameters
-                ?.Select(parameter => parameter.GetType())
-                .ToArray()
-                ?? Array.Empty<Type>();
+            Type[] argumentTypes = new Type[parameters?.Length ?? 0];
```

Comment:
```suggestion
            Type[] argumentTypes;
            if (parameters == null || parameters.Length == 0)
            {
                argumentTypes = Type.EmptyTypes;
            }
            else
            {
                argumentTypes = new Type[parameters.Length];
                for (int index = 0; index < argumentTypes.Length; index++)
                {
                    argumentTypes[index] = p[index].GetType();
                }
            }
```

Reply:
@lawrence-laz Please remove duplicate loop on line 47.

Reply:
Oops, tried out this `Commit suggestion` feature and didn't notice. Removed